contracts/token/psp721/psp721.rs

```diff
@@ -283,14 +310,22 @@ impl<T: PSP721Storage + Flush> PSP721Internal for T {
             return Err(PSP721Error::TokenExists)
         }
 
+        self._before_token_transfer(None, Some(&to), &id)?;
+
         self._add_token(to, id.clone())?;
         self._emit_transfer_event(None, Some(to), id);
+
+        self._after_token_transfer(None, Some(&to), &id)?;
         Ok(())
     }
 
     default fn _burn_from(&mut self, from: AccountId, id: Id) -> Result<(), PSP721Error> {
+        self._before_token_transfer(Some(&from), None, &id)?;
+
         self._remove_token(from, &id)?;
```

Comment:
The same here, let;s simplify the logic, you don't need to check allowance and ownershiping